### PR TITLE
Add epinowcast/actions/install-cmdstan + pgdown as env object

### PR DESCRIPTION
**Pull request by seabbs**
## Description

This PR is not linked to an issue. It overhauls the current CI to better match practice in `epinowcast` (mostly this is using our `cmdstan` action which has caching to speed up builds of CI) in order to begin looking at the issue with the `pkgdown` docs. With the same aim in mind it also adds saving the built website as an enviroment object (which is not standard `epinowcast` practice but should imo).

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by seabbs** [comment]
Its a bit in the weeds and ideally it would post a comment with a link or something but if you click on the `pkgdown` workflow and then on summary you should see something called website which is the built website.

**Comment by seabbs** [comment]
I triggered a build on `main` and yes it looks like the problem has cleared itself up≥

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -12,6 +12,10 @@ on:
   merge_group:
   workflow_dispatch:
 
+  concurrency:
```
Comment (athowes) [inline .github/workflows/R-CMD-check.yaml:15]: This groups together workflow runs and will cancel one if another on the same branch is triggered, or something like this?

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -71,11 +57,10 @@ jobs:
           needs: check
 
       - name: Install cmdstan
-        run: |
-          print(list.files("C:/rtools43"))
-          cmdstanr::check_cmdstan_toolchain(fix = TRUE)
-          cmdstanr::install_cmdstan(cores = 2, quiet = TRUE)
-        shell: Rscript {0}
+        uses: epinowcast/actions/install-cmdstan@v1
```
Comment (athowes) [inline .github/workflows/R-CMD-check.yaml:60]: This is a better version of installing `cmdstanr` in some way?

File: .github/workflows/pkgdown.yaml [hunk 3]
```diff
@@ -38,15 +38,22 @@ jobs:
...
+        uses: epinowcast/actions/install-cmdstan@v1
+        with:
+          cmdstan-version: 'latest'
+          num-cores: 2
 
       - name: Build site
         run: pkgdown::build_site_github_pages(new_process = FALSE, install = FALSE)
         shell: Rscript {0}
 
+      - name: Upload website
```
Comment (athowes) [inline .github/workflows/pkgdown.yaml:50]: So previously we didn't have this. It renders the website to a folder named `docs` which lives {somewhere} for a period of 5 days?

**Review by athowes — APPROVED** [review]
I've not used these functions before. If you have a chance, putting a brief response to my questions / clarifications would be helpful for me to learn what's going on. Other than that, happy to merge.

File: .github/workflows/R-CMD-check.yaml (hunk 2 above)
Comment (athowes) [reply .github/workflows/R-CMD-check.yaml:60]: > (mostly this is using our cmdstan action which has caching to speed up builds of CI)

So if a version of `cmdstanr` is available installed {somewhere} already then it won't try to reinstall it? Where would the version be stored?

File: .github/workflows/pkgdown.yaml (hunk 3 above)
Comment (athowes) [reply .github/workflows/pkgdown.yaml:50]: > saving the built website as an enviroment object

So I guess the {somewhere} is an "environment object" (also don't know what this is, looking online now).

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (seabbs) [reply .github/workflows/R-CMD-check.yaml:15]: yes exactly

File: .github/workflows/R-CMD-check.yaml (hunk 2 above)
Comment (seabbs) [reply .github/workflows/R-CMD-check.yaml:60]: No this is using `cmdstanr` to install and cache `cmdstan` - it gets cached using the github actions caching system which is also what is happening to R package dependencies

**Pull request merged**